rfcs/0002-upload-from-dropbox/text.md

```diff
@@ -10,126 +10,182 @@
+The `dataset_processing_status` table tracks processing as it progresses and has fields that statuses, progress, and
+messages for the upload, validation, and conversion processing steps.
 
-#### 3. Verify
+The status fields in the `dataset_processing_status` table are enums, which are [defined below](#get-datasetsdataset_idstatus).
 
-To prevent [server side request forgery](https://owasp.org/www-community/attacks/Server_Side_Request_Forgery) the
-incoming shared link will be verified before queuing for download. The links generated by DropBox are often not direct
-links to the files and need to be parsed into the correct shape for downloading. The correct URL will be generated
-as part of the validation process. The backend that runs this should also be given minimal privileges to further protect from
-server-side request forgery. How to download from a shared link on [Dropbox](https://help.dropbox.com/files-folders/share/force-download).
 
-The DropBox download link can be generated by modifying the query parameter of the shared link. A shared link from
-DropBox looks like _<https://www.dropbox.com/s/{folder_id}/{file_id}url?dl=0>_. By changing the query parameter `dl=0` to
-`dl=1`, the shared file can be downloaded.
+### Flow Description
 
-The URL will be verified it matches the expected format using regex. The URL must also refer to the correct DropBox domain. The URL will be rejected if it does not match.
+#### 1. Generate Shared Link on Dropbox Site
 
-#### 4. Upload Step Function/ Update Database
+The user must [generate](https://help.dropbox.com/files-folders/share/view-only-access) a shared link from their Dropbox.
+Shared links should only point to a single file, not a directory of files. Shared links from Dropbox should look like
 
-Once the request has been verified, the Data Portal backend will create an entry in the upload table using the
-_dataset_id_ as the key and set the status to _Waiting_. The backend will also kick off the upload step function.
-The step function will handle reties and direct responses to the correct handlers.
+`https://www.dropbox.com/s/abc123/my_dataset.h5ad?dl=1`
 
-##### Why Might the Upload Fail
+#### 2. Submit Shared Link Via DP Browser
 
-- The downloading server crashed.
-- DropBox experienced an outage.
-- The user has reached their [daily bandwidth allowance](https://www.dropbox.com/plans?trigger=nr) from their DropBox. [more info](https://help.dropbox.com/files-folders/share/banned-links).
+Per the functional requirements, the user must be authenticated, have created a collection, and agreed to DP data
+policies before being able to upload a dataset.
+Users will paste their Dropbox shared link into the DP Browser, which will then make a
+[POST request to /collections/{collection_id}/upload/link](#post-collectioncollection_iduploadlink) with the Dropbox
+link in the request body. The DP Browser can expect a response with a _dataset_id_ that identifies the new dataset and
```

Comment:
@mckinsel @Bento007 : After the `POST` returns the dataset id, can FE make a `GET` call to get dataset's name, or do we need to wait till validation completes? Thank you!

Reply:
I think it would need to wait until validation completes, as it'll be pulled out of the file with all the other metadata.

I think there's actually a slightly broader issue here. You can definitely POST a new dataset, receive the `dataset_id` in the response and then GET and get a 404; consistency is only eventual.

Reply:
Actually never mind, we'll just have to be slightly careful not to do that.

But the first point is true, we'll have to parse the file before we know the name, and we don't do that until after we're sure it's valid.